conf/nginx/lua/proxyctl.lua

```diff
@@ -62,7 +62,7 @@ if (lock_timestamp == 0) then
 
     -- Lanch project start script
     -- os.execute returs multiple values starting with Lua 5.2
-    local status, exit, exit_code = os.execute("sudo /usr/local/bin/proxyctl start $(sudo /usr/local/bin/proxyctl lookup \"" .. ngx.var.host .. "\")")
+    local status, exit, exit_code = os.execute("sudo -E /usr/local/bin/proxyctl start $(sudo /usr/local/bin/proxyctl lookup \"" .. ngx.var.host .. "\")")
```

Comment:
Is this change necessary?
If it was, then `proxyctl` would not be able to access `PROJECT_INACTIVITY_TIMEOUT`/etc. variables in use today, which is not the case.

Reply:
you are right, this is unnecessary. changes reverted.

Reply:
i am rechecked this. and i see what proxyctl executed from proxyctl.lua started from nginx user. nginx user does not have access to container variables. here is a list of variables available inside proxyctl started by proxyctl.lua:
```
SUDO_GID=101
USERNAME=root
SUDO_COMMAND=/usr/local/bin/proxyctl start project www3
USER=root
PWD=/
HOME=/root
SUDO_USER=nginx
SUDO_UID=100
MAIL=/var/mail/root
TERM=unknown
SHELL=/bin/ash
SHLVL=1
LOGNAME=root
LOG_PREFIX=[start]
PATH=/usr/local/bin:/bin:/usr/bin:/sbin:/usr/sbin
_=/usr/bin/env
```
i will take a look how to pass all container variables to nginx environment

Reply:
there is no way to pass ALL container variables into nginx context. all needed variables should be decalred in nginx config:
```
env DEBUG_LOG;
env PROJECT_INACTIVITY_TIMEOUT;
env PROJECT_DANGLING_TIMEOUT;
env PROJECT_AUTOSTART;
```
and yes, bash `-E` option is reqired for passing variables from nginx user context to sudo session.